lms/templates/courseware/progress.html

```diff
@@ -56,10 +56,9 @@ <h2 class="hd hd-2 progress-certificates-title">
                     ${_("Course Progress for Student '{username}' ({email})").format(username=student.username, email=student.email)}
                 </h2>
 
-                %if certificate_data:
                 <div class="wrapper-msg wrapper-auto-cert">
                     <div id="errors-info" class="errors-info"></div>
-                    %if passed:
+                    %if certificate_data:
                     <div class="auto-cert-message" id="course-success">
                         <div class="has-actions">
                             <% post_url = reverse('generate_user_cert', args=[unicode(course.id)]) %>
```

Comment:
Lines 69 through 76 in this file (can't comment directly on them because I didn't change them) ... 

The logic for viewing the three buttons - "request certificate", "view certificate", and "download certificate" seems to be pretty clear here, which is why I didn't change it.
1. View certificate when there is a web view url in the cert data
2. Download certificate when the status is "downloadable" and there is a download url in the cert data
3. Request certificate if the cert status is "requesting"